lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -275,14 +277,63 @@ class ConvertAten_IndexPutImplOp
 
     // Creating a tm_tensor.scatter op with the following mapping:
     // 1.) Index tensor from the `indicesList` maps to the indices in scatter
-    // op. Index tensor is expanded from 1-d to 2-d, and its element type is set
-    // to i32 as required for the scatter op.
+    // op. Index tensor is expanded from 1-d to 2-d, and if it is already 2-d
+    // then first it is collapsed to a 1-d tensor and then expanded. And its
+    // element type is set to i32 as required for the scatter op.
     // 2.) `values` is mapped to `updates` in scatter op.
     // 3.) `input` is mapped to `original` in scatter op.
     std::optional<unsigned> indexTensorRank = getTensorRank(indexTensor);
-    if (!indexTensorRank || *indexTensorRank != 1)
+    if (!indexTensorRank)
       return rewriter.notifyMatchFailure(
-          op, "unimplemented: index tensor with rank != 1 is not supported");
+          op, "expected index tensor to have a rank");
+
+    if (*indexTensorRank == 2) {
+      auto indexTensorType = indexTensor.getType().cast<BaseTensorType>();
+      SmallVector<int64_t> indexTensorShape(indexTensorType.getSizes());
+      if (indexTensorShape[0] == 1) {
+        ValueTensorType collapsedIndexTensorType = ValueTensorType::get(
+            context, llvm::makeArrayRef({indexTensorShape[1]}),
+            indexTensorType.getDtype());
+        Value torchCstZero = rewriter.create<Torch::ConstantIntOp>(
+            loc, rewriter.getI64IntegerAttr(0));
+        indexTensor = rewriter.create<AtenSqueezeDimOp>(
```

Comment:
I think we can just lower to `AtenFlattenUsingIntsOp`, and this way the implementation works for all 2-d index inputs, as well as higher ranked tensors